src/coffee/import.coffee

```diff
@@ -362,14 +370,20 @@ class Import
         when 'removeVariant' then @allowRemovalOfVariants
         else throw Error "The action '#{action.action}' is not supported. Please contact the commercetools support team!"
 
+    updateRequest = filtered.getUpdatePayload()
+
+    if publish
+      updateRequest.actions.push
+        action: 'publish'
+
     if @dryRun
       if filtered.shouldUpdate()
-        Promise.resolve "[row #{rowIndex}] DRY-RUN - updates for #{existingProduct.id}:\n#{_.prettify filtered.getUpdatePayload()}"
+        Promise.resolve "[row #{rowIndex}] DRY-RUN - updates for #{existingProduct.id}:\n#{_.prettify updateRequest}"
       else
         Promise.resolve "[row #{rowIndex}] DRY-RUN - nothing to update."
     else
       if filtered.shouldUpdate()
-        @client.products.byId(filtered.getUpdateId()).update(filtered.getUpdatePayload())
+        @client.products.byId(filtered.getUpdateId()).update(updateRequest)
         .then (result) =>
           @publishProduct(result.body, rowIndex)
```

Comment:
It is better to merge those two requests (update and publish) together because you will save one request and it is easier to handle concurrent modification error with just one request.

Also, it won't publish product again because of this condition: https://github.com/sphereio/sphere-node-product-csv-sync/pull/156/files/0c0c777f1e39df9c4e41c5378d05dbc5fce28742#diff-7ed9e504a5a8e3612516e56fc3dfd602R423